### PR TITLE
[FLINK-2645][jobmanager] forward exceptions when merging final accumlator results

### PR DESCRIPTION
**Pull request by mxm**
- fix forwarding
- add test case

### CODE REVIEW

**Comment by fhueske** [comment]
LGTM

**Comment by mxm** [comment]
Thanks for the review. I've since extended this pull request. Upon failure to deserialize the final task accumulator results, we now fail `Executions` and forward the Exception instead of just logging them.

File: flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java [hunk 1]
```diff
@@ -922,19 +922,18 @@ public boolean updateState(TaskExecutionState state) {
...
+						Map<AccumulatorRegistry.Metric, Accumulator<?, ?>> flinkAccumulators =
+							accumulators.deserializeFlinkAccumulators();
+						Map<String, Accumulator<?, ?>> userAccumulators =
+							accumulators.deserializeUserAccumulators(userClassLoader);
+						attempt.markFinished(flinkAccumulators, userAccumulators);
 					}
 					catch (Exception e) {
-						// we do not fail the job on deserialization problems of accumulators, but only log
 						LOG.error("Failed to deserialize final accumulator results.", e);
+						attempt.markFailed(new SerializedThrowable(e));
```
Comment (StephanEwen) [inline flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java]: I don't think you need a SerializedThrowable here. The exceptions are wrapped in serialized throwables when they are transported via actor messages.

**Comment by StephanEwen** [comment]
Looks good, modulo the one inline comment.

**Comment by chiwanpark** [comment]
Looks good to merge.

**Comment by fhueske** [comment]
+1, will merge this

**Pull request closed without merging**